near/src/runtime.rs

```diff
@@ -132,11 +132,16 @@ impl NightshadeRuntime {
         &self,
         shard_id: ShardId,
         block_hash: &CryptoHash,
+        last_validator_proposals: &[ValidatorStake],
         state_update: &mut TrieUpdate,
     ) -> Result<(), Box<dyn std::error::Error>> {
         let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
-        let (stake_info, validator_reward, kickout) =
-            epoch_manager.compute_stake_return_info(block_hash)?;
+        let (stake_info, validator_reward) = epoch_manager.compute_stake_return_info(block_hash)?;
+        let account_to_stake =
+            last_validator_proposals.iter().rev().fold(HashMap::new(), |mut acc, v| {
```

Comment:
Alternative that doesn't make use `.rev()` (which is O(n) memory) just override with the latest amount